Index: core/lib/checkpoint.h
===================================================================
RCS file: /cvsroot/contiki/contiki-2.x/core/lib/checkpoint.h,v
retrieving revision 1.1
diff -d -u -p -r1.1 checkpoint.h
--- core/lib/checkpoint.h	4 Feb 2009 17:49:48 -0000	1.1
+++ core/lib/checkpoint.h	26 Jan 2011 10:06:53 -0000
@@ -54,6 +54,6 @@ void checkpoint_arch_checkpoint(int fd);
 
 void checkpoint_arch_rollback(int fd);
 
-int checkpoint_arch_size();
+int checkpoint_arch_size(void);
 
 #endif /* CHECKPOINT_H */
Index: core/net/uip6.c
===================================================================
RCS file: /cvsroot/contiki/contiki-2.x/core/net/uip6.c,v
retrieving revision 1.25
diff -d -u -p -r1.25 uip6.c
--- core/net/uip6.c	4 Jan 2011 22:11:37 -0000	1.25
+++ core/net/uip6.c	26 Jan 2011 10:06:57 -0000
@@ -800,7 +800,7 @@ uip_add_rcv_nxt(u16_t n)
  * \brief Process the options in Destination and Hop By Hop extension headers
  */
 static u8_t
-ext_hdr_options_process() {
+ext_hdr_options_process(void) {
  /*
   * Length field in the extension header: length of th eheader in units of
   * 8 bytes, excluding the first 8 bytes
Index: cpu/avr/mtarch.c
===================================================================
RCS file: /cvsroot/contiki/contiki-2.x/cpu/avr/mtarch.c,v
retrieving revision 1.1
diff -d -u -p -r1.1 mtarch.c
--- cpu/avr/mtarch.c	22 Dec 2006 16:55:53 -0000	1.1
+++ cpu/avr/mtarch.c	26 Jan 2011 10:06:58 -0000
@@ -40,7 +40,6 @@
 #include <avr/interrupt.h>
 #include <stdio.h>
 #include "sys/mt.h"
-#include "dev/rs232.h"
 
 /*--------------------------------------------------------------------------*/
 void
Index: cpu/avr/dev/usb/usb_drv.h
===================================================================
RCS file: /cvsroot/contiki/contiki-2.x/cpu/avr/dev/usb/usb_drv.h,v
retrieving revision 1.4
diff -d -u -p -r1.4 usb_drv.h
--- cpu/avr/dev/usb/usb_drv.h	17 Sep 2010 21:59:09 -0000	1.4
+++ cpu/avr/dev/usb/usb_drv.h	26 Jan 2011 10:07:02 -0000
@@ -929,10 +929,10 @@ U8      host_determine_pipe_size     (U1
 void    host_disable_all_pipe        (void);
 U8      usb_get_nb_pipe_interrupt    (void);
 
-U8 usb_endpoint_wait_for_read_control_enabled();
-U8 usb_endpoint_wait_for_write_enabled();
-U8 usb_endpoint_wait_for_receive_out();
-U8 usb_endpoint_wait_for_IN_ready();
+U8 usb_endpoint_wait_for_read_control_enabled(void);
+U8 usb_endpoint_wait_for_write_enabled(void);
+U8 usb_endpoint_wait_for_receive_out(void);
+U8 usb_endpoint_wait_for_IN_ready(void);
 
 #define usb_endpoint_wait_for_read_enabled usb_endpoint_wait_for_write_enabled
 
